subsys/net/ip/net_shell.c

```diff
@@ -3202,6 +3202,99 @@ static int cmd_net_ping(const struct shell *shell, size_t argc, char *argv[])
 #endif
 }
 
+static struct net_pkt *get_net_pkt(const char *ptr_str)
+{
+	u8_t buf[sizeof(intptr_t)];
+	intptr_t ptr = 0;
+	size_t len;
+	int i;
+
+	if (ptr_str[0] == '0' && ptr_str[1] == 'x') {
+		ptr_str += 2;
+	}
+
+	len = hex2bin(ptr_str, strlen(ptr_str), buf, sizeof(buf));
+	if (!len) {
+		return NULL;
+	}
+
+	for (i = len - 1; i >= 0; i--) {
+		ptr |= buf[i] << 8 * (len - 1 - i);
+	}
+
```

Comment:
Shall we test if the pointer is an actual net_pkt from existing pools?

Reply:
> Shall we test if the pointer is an actual net_pkt from existing pools?

I was thinking about this but did not do it because:
* would mean that we create API in net_pkt.c for it
* net_pkt pool can be created also in other places than net_pkt.c so the check would fail

Checking API could be created if really needed but I propose to keep this simple atm.